api/versions.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"google.golang.org/appengine/datastore"
+)
+
+// apiVersionsHandler is responsible for emitting just the browser versions for the test runs.
+func apiVersionsHandler(w http.ResponseWriter, r *http.Request) {
+	product, err := shared.ParseProductParam(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	} else if product == nil {
+		http.Error(w, fmt.Sprintf("Invalid product param: %s", r.URL.Query().Get("product")), http.StatusBadRequest)
+		return
+	}
+
+	ctx := shared.NewAppEngineContext(r)
+	query := datastore.NewQuery("TestRun").Filter("BrowserName =", product.BrowserName)
+	queries := []*datastore.Query{query}
+	if product.BrowserVersion != "" {
+		queries = make([]*datastore.Query, 2)
+		queries[0] = query.Filter("BrowserVersion =", product.BrowserVersion).Limit(1)
+		queries[1] = shared.VersionPrefix(query, "BrowserVersion", product.BrowserVersion, false).
+			Project("BrowserVersion").
+			Distinct()
+	}
+
+	var runs shared.TestRuns
+	for _, query := range queries {
+		var someRuns shared.TestRuns
+		if _, err := query.GetAll(ctx, &someRuns); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		runs = append(runs, someRuns...)
+	}
+
+	if len(runs) < 1 {
+		w.WriteHeader(http.StatusNotFound)
+		w.Write([]byte("[]"))
+		return
+	}
+
+	versions := make([]string, len(runs))
+	for i := range runs {
+		versions[i] = runs[i].BrowserVersion
```

Comment:
Follow-up from above: This will probably repeat the one result from `queries[0]` (if there is one) because it will also appear in the result set from `queries[1]`.

Reply:
The tests cover exact matching (and pass) with `chrome-1` :)

https://github.com/web-platform-tests/wpt.fyi/pull/691/files/0086c84e37c341dbed0bedd1e1cf31275285b3d1#diff-8f642c273c44c9c74435d70606cb4148R65